### PR TITLE
Added unit tests and end2end tests for list permitted partitions

### PR DESCRIPTION
**Pull request by anilak11**
Details:

* Added some more cases in unit tests to check filter_args funtionality for list permitted partitions
* Added end2end test case to check the list permitted partitions functionality

### CODE REVIEW

File: tests/end2end/test_partition.py [hunk 1]
```diff
@@ -22,18 +22,19 @@
...
 # pylint: disable=line-too-long,unused-import
-from zhmcclient.testutils import hmc_definition, hmc_session  # noqa: F401, E501
 from zhmcclient.testutils import dpm_mode_cpcs  # noqa: F401, E501
-# pylint: enable=line-too-long,unused-import
-
+from zhmcclient.testutils import hmc_definition, hmc_session  # noqa: F401, E501
 from .utils import skip_warn, pick_test_resources, TEST_PREFIX, \
     standard_partition_props, runtest_find_list, runtest_get_properties
 
+# pylint: enable=line-too-long,unused-import
```
Comment (andy-maier) [inline tests/end2end/test_partition.py:36]: Why did you move the pylint line? It puts the import from `.utils` into the pylint section where unused import is disabled. That is only necessary for imports that provide pylint fixtures or their dependent fixtures (such as dpm_mode_cpcs, hmc_definition or hmc_session).

If the import from `.utils` has unused imports, they should be removed instead.

File: tests/end2end/test_partition.py [hunk 2]
```diff
@@ -204,3 +205,116 @@ def test_part_crud(dpm_mode_cpcs):  # noqa: F811
...
+        console = client.consoles.console
+
+        part_list = cpc.partitions.list()
+        if not part_list:
+            skip_warn("No partitions on CPC {c} managed by HMC {h} for "
+                      "the user {u}".format(c=cpc.name, h=hd.host,
+                                            u=session.userid))
+        part_list = pick_test_resources(part_list)
+
+        user_name = role_name = TEST_PREFIX + '_test_list_permitted_partitions'
```
Comment (andy-maier) [inline tests/end2end/test_partition.py:234]: This is a test about listing partitions. There is no need to create or delete any users or password rules.

File: tests/unit/zhmcclient/test_partition.py [hunk 3]
```diff
@@ -18,14 +18,14 @@
 
 from __future__ import absolute_import, print_function
 
-import re
 import copy
+import re
+
 import pytest
+from tests.common.utils import assert_resources
```
Comment (andy-maier) [inline tests/unit/zhmcclient/test_partition.py:25]: The order of imports so far was grouped into these groups, in order:
* built-in modules
* installed modules
* zhmcclient
* tests

There is no particular technical reason for that; it is just ordered from global to local.

`tests` is a bit special because it is just a local directory but occupies that name in the namespace of packages.

I prefer if we can keep this grouping.

**Review by andy-maier — CHANGES_REQUESTED** [review]
See comments

**Comment by andy-maier** [comment]
@anilak11 Please rebase the PR on the latest master to resolve the conflicts.

File: tests/end2end/test_partition.py (hunk 1 above)
Comment (anilak11) [reply tests/end2end/test_partition.py]: okay, reverted it back

File: tests/end2end/test_partition.py (hunk 2 above)
Comment (anilak11) [reply tests/end2end/test_partition.py]: As discussed, I thought of testing hmc functionality, Now I'm testing the list_permitted_partitions functionality

File: tests/unit/zhmcclient/test_partition.py (hunk 3 above)
Comment (anilak11) [reply tests/unit/zhmcclient/test_partition.py]: I just did a formatting of file on my pycharm IDE, reverted it back and will follow the above instructions from now on

**Review by andy-maier — APPROVED** [review]
